### PR TITLE
Adding support for SegmentTimeline and SegmentList

### PR DESCRIPTION
**Pull request by bahern**
Work related to #26

Added in additional structs to model SegmentTimeline and SegmentList and updated Period, AdaptationSet and Representation to include them as options.

### CODE REVIEW

File: mpd/segment.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package mpd
+
+type SegmentBase struct {
+	Initialization           *URL     `xml:"Initialization,omitempty"`
+	RepresentationIndex      *URL     `xml:"RepresentationIndex,omitempty"`
+	Timescale                *uint32  `xml:"timescale,attr,omitempty"`
+	PresentationTimeOffset   *uint64  `xml:"presentationTimeOffset,attr,omitempty"`
+	IndexRange               *string  `xml:"indexRange,attr,omitempty"`
+	indexRangeExact          *bool    `xml:"indexRangeExact,attr,omitempty"`
```
Comment (GeneticGenesis) [inline mpd/segment.go]: This is private and doesn't seem to be used anywhere, any reason?

File: mpd/segment.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	IndexRange *string `xml:"indexRange,attr,omitempty"`
+}
+
+type SegmentTimeline struct {
+	Segments []*SegmentTimelineSegment `xml:"S,omitempty"`
+}
+
+type SegmentTimelineSegment struct {
+	StartTime   *uint64 `xml:"t,attr,omitempty"`
+	Duration    uint64  `xml:"d,attr"`
```
Comment (GeneticGenesis) [inline mpd/segment.go]: Is this deliberately not a pointer?

File: mpd/segment.go (hunk 1 above)
Comment (bahern) [inline mpd/segment.go]: Typo, will fix.

File: mpd/mpd.go [hunk 3]
```diff
@@ -174,7 +171,7 @@ func (m *MPD) AddNewAdaptationSetAudio(mimeType string, segmentAlignment bool, s
 	as := &AdaptationSet{
 		MimeType:         Strptr(mimeType),
 		SegmentAlignment: Boolptr(segmentAlignment),
-		StartWithSAP:     Intptr(startWithSAP),
+		StartWithSAP:     Int64ptr(startWithSAP),
```
Comment (GeneticGenesis) [inline mpd/mpd.go]: Side note unrelated to your change: This certainly doesn't need to be an Int64

File: mpd/mpd.go [hunk 4]
```diff
@@ -57,8 +57,11 @@ type MPD struct {
 }
 
 type Period struct {
-	AdaptationSets []*AdaptationSet `xml:"AdaptationSet,omitempty"`
-	BaseURL        string           `xml:"BaseURL,omitempty"`
+	BaseURL         string           `xml:"BaseURL,omitempty"`
```
Comment (GeneticGenesis) [inline mpd/mpd.go]: *string vs string ?

Comment (bahern) [inline mpd/mpd.go]: BaseURL was a string before the changes so left it as is to avoid breaking anything referencing it.

**Comment by GeneticGenesis** [comment]
LGTM.

**Pull request merged**